lib/tapyrus/rpc/tapyrus_core_client.rb

```diff
@@ -4,6 +4,38 @@
 module Tapyrus
   module RPC
 
+    # Throw when happened anything http's error with connect to server.
+    #
+    # Almost case this exception happened from 401 Unauthorized.
+    # And also, throw by cause of other http's errors.
+    #
+    # You can get raw response as cause, like as:
+    #
+    # rescue TapyrusClientConnectionError => ex
+    #   ex.response
+    # end
+    class Error < StandardError
+      attr_reader :response
+      attr_reader :message
+
+      def initialize(response)
+        raise ArgumentError, "Must set response as cause." unless response
+        @response = response
+
+        # set message from response body or status code.
+        begin
+          @message = Tapyrus::RPC::response_body2json(response.body)['error']
+        rescue JSON::ParserError => _
+          # if RPC server don't send error message, then set http status message.
+          @message = "#{response&.code} #{response&.msg}"
+        end
+      end
+
```

Comment:
With the above changes, `@message` seems to be either String or Hash. This makes it inconvenient for clients to call this, as they have to handle it differently.